modules/cache/main.tf

```diff
@@ -64,12 +68,15 @@ resource "aws_s3_bucket_lifecycle_configuration" "build_cache_versioning" {
   }
 }
 
+# decision by user whether to use a customer managed key or not. Resource is encrypted either.
+# tfsec:ignore:aws-s3-encryption-customer-key
 resource "aws_s3_bucket_server_side_encryption_configuration" "build_cache_encryption" {
   bucket = aws_s3_bucket.build_cache.id
 
   rule {
     apply_server_side_encryption_by_default {
-      sse_algorithm = "AES256"
+      sse_algorithm     = "aws:kms"
```

Comment:
if no key is provided, is the bucket encrypted with a default key?

Reply:
Yes, that's true. From the docs: `The default aws/s3 AWS KMS master key is used if this element is absent while the sse_algorithm is aws:kms`.

But we should set the `bucket_key_enabled = true`. This will save up to 99% of the encryption cost.

Reply:
fixed